daemon/ucrednet.go

```diff
@@ -28,40 +28,54 @@ import (
 	sys "syscall"
 )
 
-var errNoUID = errors.New("no uid found")
+var errNoID = errors.New("no pid/uid found")
 
+const ucrednetNoProcess = uint64(0)
 const ucrednetNobody = uint32((1 << 32) - 1)
 
-func ucrednetGetUID(remoteAddr string) (uint32, error) {
+func ucrednetGet(remoteAddr string) (uint64, uint32, error) {
```

Comment:
Personally I'd replace all of the parsing below with a simple generic (and easier to verify) code that splits elements on ```';'``` and then parses each as a ```key=value```. Then the code below could just look up the key ```uid``` or ```pid```, convert it to a number and do what it has to do.